array-manipulation-1/src/compact.js

```diff
@@ -0,0 +1,19 @@
+/* exported compact */
+/*
+Define a function, named 'compact', that accepts one arguement, 'array'.
+  declare a variable named 'newArray' as an empty array.
+  being a loop that iterates through the entire 'array', where
+    for each index in the array:
+      Create an if statement that determines if the code contains any of the not allowed values.
+      Add the values to 'newAray'.
+    return the value of newArray.
+*/
+function compact(array) {
```

Comment:
You are on the right track but you could save yourself a lot of pain by considering that `false`, `NaN`, `0`, etc... are all considered to be false or [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) values.

So you could have a conditional that checks if an element is true or [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) and not false or [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy), you push the element at that array input your newArray. 

This will filter out any of the values mentioned in the compact.test.js file.

```
//  create empty newArray array
  // loop through array argument passed in
    // check if the array[i] is truthy  eg if (arr[i]) {
      // if truthy values, push array[i] into newArray
  // return newArray
 ```